lib/validators/index.js

```diff
@@ -0,0 +1,252 @@
+ * @param {object} options.didKeyComponents - An object with
+ *   various did components.
+ * @param {Array<string>} options.supportedMethods - A list of
+ *   did:key methods this implementation supports.
+ *
+ * @throws {Error} Throws did related errors.
+ *
+ * @returns {boolean} - Are the did components valid?
+ */
+// fixme this should probably be moved into the did-method-key
+// get function and be run before returning a didDocument
+export const validateDidRequest = ({
+  didKeyComponents,
+  supportedMethods,
+}) => {
+  const {scheme, method, version, multibase, did, isDidUrl} = didKeyComponents;
+  // this is for conformance with the did resolver spec
+  if(!supportedMethods.includes(method)) {
+    throw new didResolverErrors.MethodNotSupported(
+      `Unsupported method ${method}`);
+  }
+  _validateDid({scheme, did, isDidUrl});
+  if((isDidUrl && !_isValidDidUrl({didUrl: did}))) {
+    throw new didErrors.InvalidDidUrl(`Invalid didUrl ${did}`);
+  }
+  _validateDidKey({method, version, multibase});
+  _validateMultibaseEd25519({didKeyComponents});
+};
+
+// FIXME while we do validate public key formats we do not
+// convert to the requested public key format.
+/**
+ * Public did:keys can be represented in multiple formats.
+ * While we don't do any conversion in this library we still make
+ * the check.
+ *
+ * @param {object} options - Options to use.
+ * @param {Array<string>} options.publicKeyFormats - A list of
+ *   public key formats our did:key implementation supports.
+ * @param {object} options.didOptions - The didOptions from searchParams
+ *   and headers.
+ * @param {string} options.didOptions.publicKeyFormat - The format
+ *   the public key should be returned in.
+ * @param {string} options.didOptions.enableExperimentalPublicKeyTypes - An
+ *   option that can be passed in to allow experimental key types.
+ * @param {object} options.didDocument - The didDocument requred by the did
+ *   or didUrl.
+ *
+ * @throws {Error} Throws UnsupportedPublicKeyType or InvalidPublicKeyType.
+ *
+ * @returns {undefined} Returns on sucess.
+ */
+export const validateRequestOptions = ({
+  publicKeyFormats = [],
+  didOptions: {
+    publicKeyFormat,
+    enableExperimentalPublicKeyTypes = false
+  } = {},
+  didDocument,
+}) => {
+  // if no publicKeyFormat was in the request just skip this check
+  if(!publicKeyFormat) {
+    return;
+  }
+  // supported public key formats are set manually on config
+  if(!publicKeyFormats.includes(publicKeyFormat)) {
+    throw new didKeyErrors.UnsupportedPublicKeyType(
+      `Unsupported public key type ${publicKeyFormat}`);
+  }
+  // all of the other did methods so far are signature verification
+  if(!_convertSearchParamToBoolean({param: enableExperimentalPublicKeyTypes})) {
+    const verificationFormats = ['Multikey', 'JsonWebKey2020'];
+    //keyAgreement is an encryption verification method
+    if(didDocument.type.includes('KeyAgreementKey')) {
```

Comment:
This feels very dangerous... we shouldn't be doing substring matching on types. We should only be doing whole string matching on types.

Reply:
That seems fair so changed it here: https://github.com/digitalbazaar/bedrock-did-resolver-http/pull/3/commits/ef6af7a05a487a9008d9a5a39a6220551c160256